driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -26,6 +26,23 @@ enum capability_type
 	CAP_EFFECTIVE = 2,
 };
 
+/* COS kernels handle audit field differently, see [1]. To support both
+ * versions define COS subset of task_struct with a flavor suffix (which will
+ * be ignored during relocation matching [2]).
+ *
+ * [1]: https://chromium.googlesource.com/chromiumos/third_party/kernel/+/096925a44076ba5c52faa84d255a847130ff341e%5E%21/#F2
+ * [2]: https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/tools/lib/bpf/libbpf.c#n5347
+ */
+struct audit_task_info {
+	kuid_t			loginuid;
+	unsigned int		sessionid;
+	struct audit_context	*ctx;
+};
+
+struct task_struct___cos {
+	struct audit_task_info		*audit;
+} __attribute__((preserve_access_index));
+
```

Comment:
Yep, we need these definitions otherwise we cannot compile the modern bpf :/ the old one searches them in the headers and it is compiled directly on COS so no need for extra definitions, while the modern probe is compiled only once so we need all the extra structs/fields definitions.
BTW the right place to define them seems `struct_flavors.h` file since this is actually a struct flavor, please note that if you put it in that file you can also avoid the `__attribute__((preserve_access_index));` since it is added by default `#pragma clang attribute push(__attribute__((preserve_access_index)), apply_to = record)`

Reply:
Thanks @Andreagit97 for the explanation, appreciate it!